src/main/java/org/sqlite/SQLiteErrorCode.java

```diff
@@ -63,9 +63,56 @@
+    SQLITE_READONLY_ROLLBACK(776, "  Hot journal needs to be rolled back"),
+    SQLITE_IOERR_WRITE(778, " I/O error in the VFS layer while trying to write to a file on disk"),
+    SQLITE_CANTOPEN_FULLPATH(782, " The operating system was unable to convert the filename into a full pathname"),
+    SQLITE_CONSTRAINT_FOREIGNKEY(787, " A foreign key constraint failed"),
+    SQLITE_READONLY_DBMOVED(1032, " The database file has been moved since it was opened"),
+    SQLITE_IOERR_FSYNC(1034, " I/O error in the VFS layer while trying to flush previously written content"),
+    SQLITE_CANTOPEN_CONVPATH(1038, " cygwin_conv_path() system call failed while trying to open a file"),
+    SQLITE_CONSTRAINT_FUNCTION(1043, " Error reported by extension function"),
+    SQLITE_IOERR_DIR_FSYNC(1290, " I/O error in the VFS layer while trying to invoke fsync() on a directory"),
+    SQLITE_CONSTRAINT_NOTNULL(1299, " A NOT NULL constraint failed"),
+    SQLITE_IOERR_TRUNCATE(1546, " I/O error in the VFS layer while trying to truncate a file to a smaller size"),
+    SQLITE_CONSTRAINT_PRIMARYKEY(1555, " A PRIMARY KEY constraint failed"),
+    SQLITE_IOERR_FSTAT(1802, " I/O error in the VFS layer while trying to invoke fstat()"),
+    SQLITE_CONSTRAINT_TRIGGER(1811, " A RAISE function within a trigger fired, causing the SQL statement to abort"),
+    SQLITE_IOERR_UNLOCK(2058, " I/O error within xUnlock"),
+    SQLITE_CONSTRAINT_UNIQUE(2067, " A UNIQUE constraint failed"),
+    SQLITE_IOERR_RDLOCK(2314, " I/O error within xLock"),
+    SQLITE_CONSTRAINT_VTAB(2323, " Error reported by application-defined virtual table"),
+    SQLITE_IOERR_DELETE(2570, " I/O error within xDelete"),
+    SQLITE_CONSTRAINT_ROWID(2579, " rowid is not unique"),
+    SQLITE_IOERR_NOMEM(3082, " Unable to allocate sufficient memory"),
+    SQLITE_IOERR_ACCESS(3338, " I/O error within the xAccess"),
+    SQLITE_IOERR_CHECKRESERVEDLOCK(3594, " I/O error within xCheckReservedLock"),
+    SQLITE_IOERR_LOCK(3850, " I/O error in the advisory file locking logic"),
+    SQLITE_IOERR_CLOSE(4106, " I/O error within xClose"),
+    SQLITE_IOERR_SHMOPEN(4618, " I/O error within xShmMap while trying to open a new shared memory segment"),
+    SQLITE_IOERR_SHMSIZE(4874, " I/O error within xShmMap while trying to resize an existing shared memory segment"),
+    SQLITE_IOERR_SHMMAP(5386, " I/O error within xShmMap while trying to map a shared memory segment"),
+    SQLITE_IOERR_SEEK(5642, " I/O error while trying to seek a file descriptor"),
+    SQLITE_IOERR_DELETE_NOENT(5898, " The file being deleted does not exist"),
+    SQLITE_IOERR_MMAP(6154, " I/O error while trying to map or unmap part of the database file"),
+    SQLITE_IOERR_GETTEMPPATH(6410, " Unable to determine a suitable directory in which to place temporary files"),
+    SQLITE_IOERR_CONVPATH(6666, " cygwin_conv_path() system call failed");
 
     public final int code;
+    public final int extended;
```

Comment:
This was an incomplete attempt at maintaining backwards compatibility. I'll rework it, hopefully by Monday.

Just to make sure: we still want user code doing `SQLiteErrorCode.getErrorCode(ex.getErrorCode())` on an exception caused by an extended code like `SQLITE_BUSY_RECOVERY` get the non-extended instance (in this case`SQLITE_BUSY`) by default, right? Otherwise code that does that and compares the result with an instance of `SQLiteErrorCode` to determine how to handle the exceåtion would suddenly fail, so we should have some kind of setting for getting the extended code instances instaed, correct?